### ISSUE TITLE
When viewing an image, 'Download' does not work as expected

### ISSUE THREAD

**Issue by peterdalydickson** [zulip/zulip-desktop#917]
---

- **Operating System**:
  - [ ] Windows
  - [ ] Linux/Ubuntu
  - [x] macOS
- **Clear steps to reproduce the issue**:
  1. click on an image to view it;
  2. click the **Download** button, in top right corner
- **Relevant error messages and/or screenshots**:
The image loads full screen inside Zulip, but it doesn't download. You have to click the **Go Back** icon to recover.

**Comment by manavmehta** [comment]
Hello @peterdalydickson . Can you confirm one thing for me:
Is the setting `General` -> `Advanced` -> `Ask where to save file before donwloading` `off` ?

**Comment by akashnimare** [comment]
This works well for me -
https://cl.ly/7c9705203d0d

**Comment by timabbott** [comment]
@peterdalydickson what version do you have?  Make sure it's the latest 5.0.0.

**Comment by kiran** [comment]
I can reproduce with:
- `General -> Advanced -> Ask where to save file before downloading` off
- on 5.0.0

Here's a not-great screen recording: 
![image_download](https://user-images.githubusercontent.com/1088852/78961570-c8307480-7aa6-11ea-937c-8c7423472c6e.gif)

I suspect https://github.com/zulip/zulip-desktop/commit/a03f569af9c260a931cd71e5c3e489dfbc7a13e2

**Comment by akashnimare** [comment]
@kiran @peterdalydickson are you folks on the latest version i.e. v5.0.0? Also, curious on the macOS version. I'm getting the "save to location" dialog on clicking the `download` button.

**Comment by kiran** [comment]
@akashnimare yes, I'm on Zulip v5.0.0. You can also see the version number in the GIF above. 

I just re-downloaded the Zulip desktop app & reinstalled it, and can still reproduce this behavior.

I'm on macOS Catalina 10.15.4 (19E266).

The image opens within Zulip both with the "Ask where to save file" setting on and off.

**Comment by akashnimare** [comment]
Interesting. @manavmehta  can you reproduce this issue?

**Comment by peterdalydickson** [comment]
> Is the setting General -> Advanced -> Ask where to save file before downloading off ?

This setting is off.

> @peterdalydickson what version do you have? Make sure it's the latest 5.0.0.

The issue only started after upgrade to 5.0.0.

I'm on macOS Catlina 10.15.3.

https://www.loom.com/share/609d7837449846e1ad72c84993f9a628

**Comment by akashnimare** [comment]
I see. @peterdalydickson, @kiran can you please tell us what version of Zulip server you folks are using?

`your_zulip_server/api/v1/server_settings` and check out the "zulip_version".

**Comment by peterdalydickson** [comment]
We're on zulipchat.com

*Peter Daly-Dickson*
**

-

Signature powered by ( https://www.signitic.fr/ )

Sent via Superhuman iOS ( https://sprh.mn/?vip=peter@macanta.org )

**Comment by manavmehta** [comment]
@akashnimare Nope. It works well for me. See the attached gif
![Download](https://user-images.githubusercontent.com/31139861/79452802-5c9a5a00-8006-11ea-9edf-4b7e6e0ef88e.gif)
P.S. 10fps alert! Just wait, the download dialogue will appear in the gif.
The file is downloaded in the directory.

Also @peterdalydickson 
>You have to click the Go Back icon to recover.
>
As of now, that is necessary. ie. if the download is successful, user has to go back (or press esc)

**Comment by manavmehta** [comment]
Just for clarity: I missed the `Ask where.....` context. The setting doesn't play a role here.
Whether it asks for download location or not (the effect of setting) is applicable when you click on the name of the image only, not the lightbox download button

**Comment by downright-ux** [comment]
Tested with V5.0.0 desktop on windows 10 and mintOS 19.04. Both have the same issue when selecting download its unable to get back to the chat, unless you select back in browser or hard reload on client.
I then tested with v4.0.3 desktop and the download screen works correctly.

**Comment by andersk** [comment]
For the record, this is a server frontend bug, not a desktop bug. It happens to be masked in Chrome, but you can see the same thing happen in Firefox.

**Comment by captn3m0** [comment]
I'm facing a similar issue, where if I click download, the image opens in the whole window, and there is no button for me to go back. Pressing <kbd>Esc</kbd> does nothing, and I have to quit (Ctrl+Q) and start Zulip again.

**Comment by manavmehta** [comment]
> there is no button for me to go back.
>
The left sidebar has a back button (just above settings) that may serve till this issue is fixed

**Comment by abhigyank** [comment]
> For the record, this is a server frontend bug, not a desktop bug. It happens to be masked in Chrome, but you can see the same thing happen in Firefox.

@timabbott Should we move this to the webapp?

**Comment by captn3m0** [comment]
>     The left sidebar has a back button

Realized now that I have the left sidebar hidden, so as a workaround - <kbd>Ctrl+Shift+S</kbd> to show the sidebar, and then going back works.
